packages/video_player/video_player_android/pigeons/messages.dart

```diff
@@ -24,6 +24,21 @@ class LoopingMessage {
   bool isLooping;
 }
 
+class IsSupportedMessageResponse {
+  IsSupportedMessageResponse(this.isSupported);
+  bool isSupported;
+}
+
+class IsCacheSupportedMessage {
+  IsCacheSupportedMessage(this.uri);
+  String uri;
+}
+
+class ClearCacheMessageResponse {
+  ClearCacheMessageResponse(this.hasSucceeded);
+  bool hasSucceeded;
+}
```

Comment:
Oke I'll update the Pigeon version to the latest and implement the changes

Reply:
@stuartmorgan, I've updated Pigeon, but I don't exacly understand what you mean. I understand that Pigeon supports primitive returns, however, by adding these primitives to a "class" it makes it more flexible when we want to add additional properties to the class in the future without changing too much. Let me know if I'm correct here or if you would like the modifications.

Reply:
> by adding these primitives to a "class" it makes it more flexible when we want to add additional properties to the class in the future without changing too much

This is a pattern we follow for platform interface methods because those are public API intended for subclassing, so adding parameters there is breaking. The Pigeon interface is entirely internal, so if we wanted additional parameters we would just add them, and in the unlikely event we needed additional return data in a response we would add a method at that point. There's no reason for Pigeon methods to pre-emptively use classes for everything any more than any other internal API.